server/v2/cometbft/abci.go

```diff
@@ -79,7 +80,13 @@ func (c *Consensus[T]) CheckTx(ctx context.Context, req *abci.RequestCheckTx) (*
 	if err != nil {
 		return nil, err
 	}
-	resp, err := c.app.ValidateTx(ctx, decodedTx)
+	var execMode corecontext.ExecMode
```

Comment:
nit, the default value will be already `corecontext.ExecModeCheck` so maybe we can drop the else and do `execMode := corecontext.ExecModeCheck` so that it is explicit.

Reply:
makes sense, ill adjust